### PR TITLE
feat: add external service timeout settings in frontend

### LINKED ISSUE #6041: Allow default connect/read timeout and more to be customizable
## Abstract

At the moment all connection related timeout and retries variables are hard-coded constants in the code:
https://github.com/rotki/rotki/blob/9e640017554fd44f7eedda06ca79b8edfa344105/rotkehlchen/constants/timing.py#L16-L18

There is no possibility to customize them depending on the user's situation and connection capabilities.

## Motivation

If a user is in a slow connection place we should allow them to configure these variables.

## Specification

Make them configurable settings

### PR DESCRIPTION
**Pull request by lukicenturi**
Closes #6041
## Checklist

- [ ] The PR modified the frontend, and updated the [user guide](https://github.com/rotki/rotki/blob/develop/docs/usage_guide.rst) to reflect the changes.

### CODE REVIEW

File: frontend/app/src/components/settings/general/external-services/ReadTimeoutSetting.vue [hunk 1]
```diff
@@ -0,0 +1,78 @@
...
+        persistent-hint
+        type="number"
+        :min="minPeriod"
+        :success-messages="success"
+        :error-messages="error || v$.readTimeout.$errors.map(e => e.$message)"
+        @change="callIfValid($event, update)"
+      />
+    </SettingsOption>
+  </div>
+</template>
```
Comment (kelsos) [inline frontend/app/src/components/settings/general/external-services/ReadTimeoutSetting.vue:78]: These 3 components are quite similar can't we extract a component that shares the common code?

File: frontend/app/src/locales/en.json [hunk 2]
```diff
@@ -2330,6 +2330,31 @@
         "empty": "The decimal separator cannot be empty"
       }
     },
+    "external_service_setting": {
+      "label": {
+        "query_retry_limit": "Query retry limit",
+        "query_retry_limit_hint": "The number of times to retry a query to external services before giving up.",
```
Comment (kelsos) [inline frontend/app/src/locales/en.json:2336]: Any way we could at least translate these new entries to `cn` too?

**Review by kelsos — COMMENTED** [review]
left some initial comments

File: frontend/app/src/locales/es.json [hunk 3]
```diff
@@ -2222,6 +2222,31 @@
         "empty": "The decimal separator cannot be empty"
       }
     },
+    "external_service_setting": {
+      "label": {
+        "query_retry_limit": "查询重试限制",
```
Comment (kelsos) [inline frontend/app/src/locales/es.json:2227]: ¡Un momento! ¡Esto no parece español! (used deepl for this)
Wait a minute!!! this does not look like Spanish! 🇪🇸 🤣 

This we will have to revert

File: frontend/app/src/locales/gr.json [hunk 4]
```diff
@@ -2227,6 +2227,31 @@
...
+        "connect_timeout": {
+          "non_empty": "请输入连接超时",
+          "invalid_range": "连接超时必须大于{min}"
+        },
+        "read_timeout": {
+          "non_empty": "请输入读取超时",
+          "invalid_range": "读取超时必须大于{min}"
+        }
+      }
+    },
```
Comment (kelsos) [inline frontend/app/src/locales/gr.json:2254]: We should certainly not add chinese translation to other languages except the cn one right?

File: frontend/app/src/locales/cn.json [hunk 5]
```diff
@@ -2265,6 +2265,31 @@
         "empty": "小数分隔符不能为空"
       }
     },
+    "external_service_setting": {
```
Comment (kelsos) [inline frontend/app/src/locales/cn.json:2268]: The auto-translation says this is alright, thank you 👍

File: frontend/app/src/locales/gr.json (hunk 4 above)
Comment (lukicenturi) [reply frontend/app/src/locales/gr.json:2254]: oops, should copy the `en` version instead.

File: frontend/app/src/locales/es.json (hunk 3 above)
Comment (lukicenturi) [reply frontend/app/src/locales/es.json]: my bad, should just let chatgpt handle it 😅

**Review by kelsos — APPROVED** [review]
lgtm @lukicenturi thanks

**Pull request merged**